From 9f63343b5a7ef0ef4d83f8cbb21618ccfb8dbdb7 Mon Sep 17 00:00:00 2001
From: Rajendra Nayak <rnayak@ti.com>
Date: Thu, 20 Jun 2013 16:28:54 +0530
Subject: [PATCH 0365/1115] ARM: DRA7: PM: Add mpu <--> emif staticdep

EMIF to EMPU static dependency is needed on DRA7 similar
to OMAP5 because the dynamic dependencies are broken.

Signed-off-by: Rajendra Nayak <rnayak@ti.com>
(cherry picked from commit 02c829156e26ed0124877b978553877be187aab2)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-omap2/pm44xx.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/pm44xx.c b/arch/arm/mach-omap2/pm44xx.c
index 7491742..3dbfa77 100644
--- a/arch/arm/mach-omap2/pm44xx.c
+++ b/arch/arm/mach-omap2/pm44xx.c
@@ -202,14 +202,15 @@ static inline int omap4_init_static_deps(void)
 }
 
 /**
- * omap5_init_static_deps - Init static clkdm dependencies on OMAP5
+ * omap5_dra7_init_static_deps - Init static clkdm dependencies on OMAP5
+ * and DRA7
  *
  * The dynamic dependency between MPUSS -> EMIF is broken and has
  * not worked as expected. The hardware recommendation is to
  * enable static dependencies for these to avoid system
  * lock ups or random crashes.
  */
-static inline int omap5_init_static_deps(void)
+static inline int omap5_dra7_init_static_deps(void)
 {
 	struct clockdomain *mpuss_clkdm, *emif_clkdm;
 	int ret;
@@ -252,8 +253,8 @@ int __init omap4_pm_init(void)
 
 	if (cpu_is_omap44xx())
 		ret = omap4_init_static_deps();
-	else if (soc_is_omap54xx())
-		ret = omap5_init_static_deps();
+	else if (soc_is_omap54xx() || soc_is_dra7xx())
+		ret = omap5_dra7_init_static_deps();
 
 	if (ret) {
 		pr_err("Failed to initialise static dependencies.\n");
-- 
1.7.5.4

